src/logic/hooks/useIsSmartContractWallet.ts

```diff
@@ -0,0 +1,18 @@
+import { useSelector } from 'react-redux'
+
+import { currentChainId } from 'src/logic/config/store/selectors'
+import { isSmartContractWallet } from 'src/logic/wallets/getWeb3'
+import useAsyncValue from 'src/logic/hooks/useAsyncValue'
+
+const useIsSmartContractWallet = (userAccount: string): boolean => {
+  const chainId = useSelector(currentChainId)
+
+  // chainId is not an argument of isSmartContractWallet, but it is used as
+  // dependency. isSmartContract calls getWeb3ReadOnly(), which relates to the chainId
+  const result = useAsyncValue<boolean>(isSmartContractWallet, () => [userAccount, chainId])
```

Comment:
Looks great! A final cherry on top would be to use a memoized version of `isSmartContractWallet` here.

Reply:
We would probably have to send the web3 object to `isSmartContractWallet` as well, or some indicator for `chainId` changing? If we memoized based on just the `account`, it would not be able to verify when switching chains as the address would still remain the same.